torch/linalg/__init__.py

```diff
@@ -204,31 +213,42 @@
 slogdet = _add_docstr(_linalg.linalg_slogdet, r"""
 linalg.slogdet(input, *, out=None) -> (Tensor, Tensor)
 
-Calculates the sign and natural logarithm of the absolute value of a square matrix's determinant,
-or of the absolute values of the determinants of a batch of square matrices :attr:`input`.
-The determinant can be computed with ``sign * exp(logabsdet)``.
+Computes the sign and logarithm of the norm of the determinant of a square matrix or batch of square matrices.
 
-Supports input of float, double, cfloat and cdouble datatypes.
+For a complex :attr:`input`, as the determinant is complex in this case, it returns the angle
+and the logarithm of the modulus of the determinant.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+The returend value decomposition is represented as a namedtuple `(sign, logabsdet)`.
+If :attr:`input` is a batch of matrices, then `sign`, `logabsdet` are also batched with the same
+batch dimensions. The determinant can be recovered as `sign * exp(logabsdet)`.
 
-.. note:: The determinant is computed using LU factorization. LAPACK's `getrf` is used for CPU inputs,
-          and MAGMA's `getrf` is used for CUDA inputs.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+`logabsdet` will always be real-valued, even if :attr:`input` is complex.
```

Comment:
This seems like it should be connected with the prior discussion of sign and logabsdet. Logically this function's documentation jumps around a lot.

I would expect notes to generally go after regular documentation, and the "supports" statement to go after the opening definition + mathematical statement (if any). Then this discussion of the return values could precede the notes, maybe?

Reply:
In all the functions I tried to follow the following schema (open to discussion about whether it's reasonable or there are better ones):
- One line summary
- Mathematical statement of what it does. In LaTeX if it's non-trivial (e.g. for factorisations). Here both the real and complex cases are treated.
- Describe the return if it's a tuple. Put the return size if it is non-trivial (e.g. describe it for `qr` but not for `norm` or `inv`). 
- Describe the arguments and what they do. Sometimes the size of the return is described here if the size of the return depends on some arguments
- Describe the supported dtypes and the dtype of the outputs if it's not the same as that of the inputs (as in this case).

It could be argued that that last step could be moved up. I would not mind to do so.

I have avoided putting too much information in the `Arguments` section, and I have prefered to completely avoid using the `Returns` section. The reasoning behind this is that, in my experience, they are too far back in the documentation. Furthermore, when you need to describe some behaviour that is particularly tricky (e.g. the norms in `linalg.norm` / `linalg.cond`) the indentation makes everything way too crammed.